File: python_files/raylib/pong/ball.py
import pyray as p


class Ball:
    def __init__(self) -> None:
        self.radius = 10
        self.initial_x = p.get_screen_width() / 2
        self.initial_y = p.get_screen_height() / 2
        self.x = self.initial_x
        self.y = self.initial_y
        self.color = p.RED
        self.speed_x = 5
        self.speed_y = 6
        self.frames_counter = 0

    def reset(self):
        self.x = self.initial_x
        self.y = self.initial_y

    def draw(self):
        p.draw_circle_v((self.x, self.y), self.radius, self.color)

    def update(self):
        self.frames_counter += 1
        if self.frames_counter > 60:
            self.frames_counter = 61
            self.x += self.speed_x
            self.y += self.speed_y
        

        if self.y <= self.radius or self.y >= p.get_screen_height() - self.radius:
            self.speed_y *= -1

File: c_files/raylib/pong/ball.c
#include "raylib.h"

#include "ball.h"

void ball_initialize(Ball *ball)
{
    ball->initial_x = (float)GetScreenWidth() / 2;
    ball->initial_y = (float)GetScreenHeight() / 2;
    ball->x = ball->initial_x;
    ball->y = ball->initial_y;
    ball->radius = 10;
    ball->color = RED;
    ball->initial_speed_x = 5;
    ball->initial_speed_y = 5;
    ball->speed_x = ball->initial_speed_x;
    ball->speed_y = ball->initial_speed_y;
    ball->speed_increment = 0.005;
    ball->is_active = false;
}

void ball_reset(Ball *ball)
{
    ball->x = ball->initial_x;
    ball->y = ball->initial_y;
    ball->speed_x = ball->initial_speed_x;
    ball->speed_y = ball->initial_speed_y;
    ball->speed_x *= -1;
    ball->is_active = false;
}

void ball_draw(Ball *ball)
{
    DrawCircleV((Vector2){ball->x, ball->y}, ball->radius, ball->color);
}

void ball_update(Ball *ball)
{
    if (IsKeyDown(KEY_SPACE))
    {
        ball->is_active = true;
    }

    if (ball->is_active)
    {
        ball->x += ball->speed_x;
        ball->y += ball->speed_y;

        ball->speed_x += ball->speed_increment;
        ball->speed_y += ball->speed_increment;
    }

    if (ball->y <= ball->radius || ball->y >= GetScreenHeight() - ball->radius)
    {
        ball->speed_y *= -1;
    }
}